packages/ag-charts-community/src/chart/interaction/interactionManager.ts

```diff
@@ -147,12 +143,31 @@ export class InteractionManager extends BaseManager<
         }
     }
 
-    resume(pauseType: PauseType) {
-        this.pausers[pauseType]--;
+    // Wrapper to only broadcast events when the InteractionManager is a given state.
+    override addListener<T extends InteractionTypes>(
+        type: T,
+        handler: (event: InteractionEvent<T> & { type: T }) => void,
+        triggeringStates: InteractionState = InteractionState.Default
+    ) {
+        return super.addListener(type, (e) => {
+            if (this.state & triggeringStates) {
+                handler(e);
+            }
+        });
     }
 
-    pause(pauseType: PauseType) {
-        this.pausers[pauseType]++;
+    public pushState(state: InteractionState) {
+        this.stateQueue.add(state);
+        this.updateCurrentState();
+    }
+
+    public popState(state: InteractionState) {
+        this.stateQueue.delete(state);
+        this.updateCurrentState();
+    }
+
+    private updateCurrentState() {
+        this.state = Array.from(this.stateQueue).reduce((max, s) => Math.max(max, s), InteractionState.Default);
     }
```

Comment:
```suggestion
    public pushState(state: InteractionState) {
        this.state |= state;
    }

    public popState(state: InteractionState) {
        this.state &= state;
    }
```

Going over the entire PR again, I wonder if we really need the `stateQueue` property. I think that since we're using bitwise flags it's unnecessary.
It does however means that it's better if we set the `Default` state to `0` because technically `Default` should mean "no specific state was chosen".

Reply:
The `stateQueue` is necessary to facilitate coordination. Intuitively, we might want to use simple setters & getters instead, but imagine this possible use-case:

1. State is initialised to `Default`.
2. AnimationManager starts and sets the state to `Animation`.
3. During the animation, a `'content-menu'` event is fired.
4. The context menu module hears this event, sets the state to `ContextMenu` and the `AnimationManager` to stop.
5. AnimationManager stops and removes the `Animation` state.

The key challenge is that after step 5, we want the state to be `ContextMenu` and not `Default`. The AnimationManager would need extra logic to figure out which state to transition to when the animation is completed or canceled.

Using the `InteractionState` values as priority number simplifies this problem significantly, because we just need to calculate the maximum to figure out the current state.

That said however, I agree that using a `Set` is unnecessary. We could just store the same information of `stateQueue` in a `number`.

Reply:
`Default` cannot be `0` because if it is then listeners will not be able to specify whether or not they want to watch this state. Example:

```
const { Default, Animation } = InteractionState;
interactionManager.addListener('hover', onHover, Default);
interactionManager.addListener('drag', onHover, Default & Animation);
```

For `'hover'` events, the bitwise & on `Default` will always be 0 if `Default=0`, so the `onHover` listener will never receive events.

Reply:
Done. The `stateQueue` is now a bitfield rather than a `Set`.